src/overlays/actors/ovl_Bg_Ikninside/z_bg_ikninside.c

```diff
@@ -28,30 +30,126 @@ const ActorInit Bg_Ikninside_InitVars = {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    } else if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->actionFunc = func_80C07220;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void func_80C072D0(BgIkninside* this, GlobalContext* globalCtx) {
+    s16 altitude;
+    s16 azimuth;
+    Vec3f pos;
+    Vec3f velocity;
+    f32 speed;
+    s32 i;
+
+    if (this->collider.base.acFlags & AC_HIT) {
+        if ((this->collider.info.acHitInfo != NULL) && (this->collider.info.acHitInfo->toucher.dmgFlags & 0x80000000)) {
+            for (i = 0; i < 20; i++) {
+                altitude = Rand_S16Offset(0x1800, 0x2800);
+                azimuth = (u32)Rand_Next() >> 0x10;
+                speed = Rand_ZeroFloat(3.0f) + 8.0f;
+                velocity.x = speed * Math_CosS(altitude) * Math_SinS(azimuth);
+                velocity.y = speed * Math_SinS(altitude) + Rand_ZeroFloat(5.0f);
+                velocity.z = speed * Math_CosS(altitude) * Math_CosS(azimuth);
+                pos.x = Rand_ZeroFloat(10.0f) * velocity.x + this->dyna.actor.world.pos.x;
+                pos.y = Rand_ZeroFloat(1.0f) * velocity.y + this->dyna.actor.world.pos.y;
+                pos.z = Rand_ZeroFloat(10.0f) * velocity.z + this->dyna.actor.world.pos.z;
+                EffectSsHahen_Spawn(globalCtx, &pos, &velocity, &D_80C076D4, 0, 30, OBJECT_IKNINSIDE_OBJ, 25,
+                                    D_80C076A0[i & 1]);
+            }
+            Flags_SetSwitch(globalCtx, DMIKNINSIDE_GET_SWITCH(&this->dyna.actor));
+            this->actionFunc = func_80C07230;
+            this->dyna.actor.draw = NULL;
+            func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+        } else {
+            this->unk_1AA = 20;
+        }
+    }
+    if (this->unk_1AA > 0) {
+        if ((this->unk_1AA % 2) != 0) {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 1.0f;
+        } else {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y;
+        }
+        this->unk_1AA--;
+    } else {
+        this->unk_1AA = 0;
+        this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y;
+    }
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void BgIkninside_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkninside* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void BgIkninside_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPDisplayList(POLY_OPA_DISP++, object_ikninside_obj_DL_00CC78);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
```

Comment:
```suggestion
    OPEN_DISPS(globalCtx->state.gfxCtx);

    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
    func_8012C28C(globalCtx->state.gfxCtx);
    gSPDisplayList(POLY_OPA_DISP++, object_ikninside_obj_DL_00CC78);

    CLOSE_DISPS(globalCtx->state.gfxCtx);
```
It's a bit easier to read if you separate the OPEN and CLOSE macros.